mobile/src/androidTest/java/com/novoda/canvas/BreadCatTest.java

```diff
@@ -0,0 +1,109 @@
+package com.novoda.canvas;
+
+import com.novoda.canvas.base.NovodaActivityTest;
+
+import android.animation.Animator;
+import android.app.Activity;
+import android.view.ViewGroup;
+import android.view.animation.DecelerateInterpolator;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+
+public class BreadCatTest extends NovodaActivityTest {
+
+    // Data
+    int[] cats = new int[]{
+        R.drawable.cat1,
+        R.drawable.cat2,
+        R.drawable.cat3,
+        R.drawable.cat4,
+        R.drawable.cat5,
+        R.drawable.cat6
+    };
+
+    // Config
+    final static int ANIMATION_DURATION = 100;
+    final static int STILL_DURATION = 500;
+
+    // Objects
+    ImageView imageBottom, imageTop;
+    int currentCat;
+
+    @Override
+    public void startTestFor(final Activity activity) {
+        currentCat = 0;
+
+        imageBottom = new ImageView(activity);
+        imageTop = new ImageView(activity);
+
+        imageBottom
+            .setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.MATCH_PARENT));
+        imageTop.setLayoutParams(new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,
+            ViewGroup.LayoutParams.MATCH_PARENT));
+
+        imageBottom.setAdjustViewBounds(true);
+        imageBottom.setScaleType(ImageView.ScaleType.CENTER_CROP);
+        imageTop.setAdjustViewBounds(true);
+        imageTop.setScaleType(ImageView.ScaleType.CENTER_CROP);
+
+        getParent().addView(imageBottom);
+        getParent().addView(imageTop);
+
+        lockAndLoad();
+    }
+
+    private void setNewCat() {
+        for(int i = 0; i < 10; i ++) {
+            int randomCat = getRandomCat();
+            if(randomCat != currentCat) {
```

Comment:
interesting check :) the probability is low right and even if it did equal it then resetting the cat isn't that big a change :dancer: